SVT Geometry Factory

pejdavies1606 opened the pull request:
Hi Andrey,

I've ported my code to use the libraries in JCSG. I had to add a few new methods that existed in Gagik's code but not in yours. I've added a G4Tubs class, and the corresponding Tube primitive, but I've left the toPolygons method empty.

How do I import this new code into the groovy scripts for gemc? Is there a jar to build?

Here's a ROOT script you can use to check the GDML output generated by running org.jlab.detector.geant4.v2.SVT.main.java. In main.java, flip the comments on lines 38 and 39 to see the whole thing or just one sector. Also, flip the comment on line 58 to see the sensor strips.
[drawIdeal.txt](https://github.com/drewkenjo/JCSG/files/579457/drawIdeal.txt)

I also fixed a few typos. :)

Commit:
added SVTConstants from CLAS12-SVT-GEOMETRY package

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
moved SVTGeant4Factory back

Commit:
moved SVTGeant4Factory back

Commit:
added .gitignore

Commit:
v1.0.0
Started porting SVT geometry from coatjava

Commit:
v1.0.0
Started adding classes for G4Tubs for Tube Primitive.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
v1.0.0
Finished rewriting methods to add functionality to Vector3d, Triangle3d, and Geant4Basic
Added G4Tubs and Tube Primitive

Commit:
v1.0.0
Added SVTStripFactory

Commit:
v1.0.1
Finally got gradle to compile with javafx on oracle.
Fixed some runtime errors.
Bodge: Copied toPolygons method from Box.java to Tube.java.

Commit:
v1.0.0
To Do: toPolygons method in Tube.java currently just returns an empty ArrayList to not cause a NullPointerException.

Commit:
v1.0.2
added volumes for epoxy and high voltage rail and pads
added author documentation

Commit:
Merge remote-tracking branch 'upstream/master'

drewkenjo commented:
Hi Peter,
Good job! Could you please move all components into SVT folder? It would 
encapsulate all Alignment, Misc etc classes in SVT package.
Thanks,
Andrey.

On 11/08/2016 04:59 PM, Peter EJ Davies wrote:

> Hi Andrey,
> 
> I've ported my code to use the libraries in JCSG. I had to add a few 
> new methods that existed in Gagik's code but not in yours. I've added 
> a G4Tubs class, and the corresponding Tube primitive, but I've left 
> the toPolygons method empty.
> 
> How do I import this new code into the groovy scripts for gemc? Is 
> there a jar to build?
> 
> Here's a ROOT script you can use to check the GDML output generated by 
> running org.jlab.detector.geant4.v2.SVT.main.java.
> drawIdeal.txt 
> https://github.com/drewkenjo/JCSG/files/579457/drawIdeal.txt
> 
> I also fixed a few typos. :)
> 
> ---
> 
> ```
>     You can view, comment on, or merge this pull request online at:
> ```
> 
> https://github.com/drewkenjo/JCSG/pull/1
> 
> ```
>     Commit Summary
> ```
> - added SVTConstants from CLAS12-SVT-GEOMETRY package
> - Merge remote-tracking branch 'upstream/master'
> - moved SVTGeant4Factory back
> - moved SVTGeant4Factory back
> - added .gitignore
> - v1.0.0
> - v1.0.0
> - Merge remote-tracking branch 'upstream/master'
> - v1.0.0
> - v1.0.0
> - v1.0.1
> - v1.0.0
> - v1.0.2
> - Merge remote-tracking branch 'upstream/master'
>   
>   ```
>   File Changes
>   ```
> - _A_ .classpath
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-0 (16)
> - _M_ .gitignore
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-1 (1)
> - _A_ .project
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-2 (23)
> - _M_ src/main/java/eu/mihosoft/vrl/v3d/Primitive.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-3 (2)
> - _M_ src/main/java/eu/mihosoft/vrl/v3d/Vector3d.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-4 (38)
> - _A_
>   src/main/java/org/jlab/detector/geant4/v2/Alignment/AlignmentFactory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-5 (438)
> - _A_ src/main/java/org/jlab/detector/geant4/v2/Misc/Matrix.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-6 (375)
> - _A_ src/main/java/org/jlab/detector/geant4/v2/Misc/Util.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-7 (661)
> - _A_
>   src/main/java/org/jlab/detector/geant4/v2/SVT/SVTAlignmentFactory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-8 (374)
> - _A_
>   src/main/java/org/jlab/detector/geant4/v2/SVT/SVTConstants.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-9 (793)
> - _A_
>   src/main/java/org/jlab/detector/geant4/v2/SVT/SVTStripFactory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-10 (506)
> - _A_
>   src/main/java/org/jlab/detector/geant4/v2/SVT/SVTVolumeFactory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-11 (1119)
> - _A_ src/main/java/org/jlab/detector/geant4/v2/SVT/main.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-12 (129)
> - _M_
>   src/main/java/org/jlab/detector/geant4/v2/SVTGeant4Factory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-13 (1)
> - _A_ src/main/java/org/jlab/detector/volume/G4Tubs.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-14 (23)
> - _M_ src/main/java/org/jlab/detector/volume/Geant4Basic.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-15 (20)
> - _A_ src/main/java/org/jlab/geometry/exporter/GdmlFile.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-16 (867)
> - _A_ src/main/java/org/jlab/geometry/exporter/IGdmlExporter.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-17 (14)
> - _A_ src/main/java/org/jlab/geometry/exporter/IVolumeExporter.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-18 (12)
> - _A_
>   src/main/java/org/jlab/geometry/exporter/VolumeExporterFactory.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-19 (18)
> - _M_ src/main/java/org/jlab/geometry/prim/Box.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-20 (2)
> - _M_ src/main/java/org/jlab/geometry/prim/Triangle3d.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-21 (28)
> - _A_ src/main/java/org/jlab/geometry/prim/Tube.java
>   https://github.com/drewkenjo/JCSG/pull/1/files#diff-22 (50)
>   
>   ```
>   Patch Links:
>   ```
> - https://github.com/drewkenjo/JCSG/pull/1.patch
> - https://github.com/drewkenjo/JCSG/pull/1.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> https://github.com/drewkenjo/JCSG/pull/1, or mute the thread 
> https://github.com/notifications/unsubscribe-auth/AMmDewLdG40avpBqxOd8skNrfwBDvdTRks5q8PDLgaJpZM4Ks9XR.

Commit:
v1.0.4
moved new classes all under SVT package
fixed arrow volumes
added theta and phi methods to Vector3d

pejdavies1606 commented:
Hi Andrey,
v1.0.4 now has all new classes under the SVT package.
~Peter

Commit:
SVT v1.0.5
started adding downstream support

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
SVT v1.0.6
fixed bug in createFiducial of SVTAlignmentFactory

Commit:
SVT v1.0.6
tweaked name of gdml exporter interface

Commit:
SVT v1.0.7

Commit:
SVT v1.0.8
Fixed volume placement

Commit:
SVT v1.0.9
restored nested structure of sector volumes
cleaned verbose output

Commit:
SVT added hidden files to .gitignore

Commit:
SVT properly added hidden files to gitignore

Commit:
SVT removed gdml exporter package

Commit:
Merge branch 'fix'

Commit:
SVT v1.0.10
removed argument from SVTConstants.connect
fixed bug with pads in SVTVolumeFactory

Commit:
SVT v1.0.11
changed how load in SVTConstants is called
started debugging rotations in applyShift(volume) of AlignmentFactory
